src/main/java/org/spongepowered/common/mixin/core/world/chunk/ChunkMixin.java

```diff
@@ -584,8 +576,22 @@ public IBlockState setBlockState(final BlockPos pos, final IBlockState state) {
                     }
                     this.world.setTileEntity(pos, tileentity);
                 }
+                // Sponge end
             }
 
+            // Sponge start - enqueue any changes to the transaction
+            if (transaction != null) {
+                // Go ahead and log the tile being replaced, the tile being removed will be at least already notified of removal
+                transaction.queueTileSet = tileentity;
+                if (tileentity != null) {
+                    ((TileEntityBridge) tileentity).bridge$setCaptured(true);
+                    tileentity.setWorld(this.world);
+                    tileentity.setPos(pos);
+                }
+                transaction.enqueueChanges(mixinWorld.bridge$getProxyAccess(), peek.getCapturedBlockSupplier());
```

Comment:
Reasoning why this was above was due to the expectations of what the "unchanged" original intentions would otherwise perform. I'm not 100% sure that the side effectual cases that are with the tile entity being potentially null here (since the if statement that this check/logic was otherwise taken from) already performs that check.